lib/order.rb

```diff
@@ -0,0 +1,53 @@
+require_relative 'menu'
+require_relative 'sms'
+
+class Order
+  attr_reader :basket
+
+  def initialize 
+    @basket = []
+    #eventually would be equal to a premade menu object (i.e Italian_restaurant) or would be able to select a premade menu
+    @menu = Menu.new   
+  end
+
+  def add_dish_to_basket(dish)
+    count = 0
+    @menu.menu.each { |food| @basket << food && count +=1 if food.name == dish }
```

Comment:
This could be simplified by - for example - adding a method `find_dish` on the `Menu` class which would find one dish by name (or return `nil`), and so we could now simply do:
```ruby
dish = @menu.find_dish(dish_name)
raise 'There is no such dish on the menu' if dish.nil? 
@basket << dish
```